app/forms/destination/schedule_rate_changes_form.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Destination
+  class ScheduleRateChangesForm < ApplicationForm
+    attribute :apply_time, :date
+    attribute :initial_interval, :integer
+    attribute :initial_rate, :decimal
+    attribute :next_interval, :integer
+    attribute :next_rate, :decimal
+    attribute :connect_fee, :decimal
+    attribute :ids_sql, :string
+
+    validates :initial_rate, :next_rate, :connect_fee, numericality: true
+    validates :initial_interval, :next_interval, numericality: { greater_than: 0 }
+    validates :ids_sql, :apply_time, presence: true
+    validate :validate_apply_time
+
+    def self.form_data
+      {
+        apply_time: 'datepicker',
+        initial_interval: 'text',
+        initial_rate: 'text',
+        next_interval: 'text',
+        next_rate: 'text',
+        connect_fee: 'text'
```

Comment:
Also maybe all inputs except `apply_time` should be `:number`?

Reply:
It's looks like active_admin_scoped_collection_actions does not supported 'number' for inputs
https://github.com/activeadmin-plugins/active_admin_scoped_collection_actions/blob/master/vendor/assets/javascripts/lib/dialog_mass_fields_update.js
```js
    if (/^(datepicker|checkbox|text)$/.test(type)) {
      wrapper = 'input';
    } else if ($.isArray(type)) {
      [wrapper, elem, opts, type] = Array.from(['select', 'option', type, '']);
    } else {
      throw new Error(`Unsupported input type: {${name}: ${type}}`);
    }
```